### PR TITLE
Adds new-bathroom react form w/ testing

### PR DESCRIPTION
**Pull request by nickbreid**
-New bathroom form on index
- create action in bathroom controller
- appropriate testing for everything!

### CODE REVIEW

File: app/controllers/api/v1/bathrooms_controller.rb [hunk 1]
```diff
@@ -20,8 +22,24 @@ def show
           }
         end
 
+
         render json: { status: 'SUCCESS', message: 'Loaded bathrooms', bathrooms: bathroom, reviews: parsed_reviews }, status: :ok
       end
+
+      def create
+        bathroom = current_user.bathrooms.new(bathroom_params)
```
Comment (chrissyfancy) [inline app/controllers/api/v1/bathrooms_controller.rb:30]: Can any users create a new bathroom or do you just want this limited to Users in your database?

File: app/javascript/react/src/containers/BathroomForm.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          name='gender'
+          label='Gender'
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.gender}
+        />
+        <QuantitySelector
+          name="toiletQuantity"
+          formFieldChange={this.handleChange}
+          fieldContent={this.state.toiletQuantity}
+        />
```
Comment (chrissyfancy) [inline app/javascript/react/src/containers/BathroomForm.js:160]: Another way to cut down on repetition would be to make a hash and loop through the values to return one FormField each time

File: spec/controllers/bathrooms_controller_spec.rb [hunk 3]
```diff
@@ -14,7 +14,69 @@
       expect(response.status).to eq 200
       expect(response.content_type).to eq("application/json")
       expect(first["establishment"]).to eq "McDonalds"
+    end
+  end
+
+end
```
Comment (chrissyfancy) [inline spec/controllers/bathrooms_controller_spec.rb:20]: Remove unnecessary empty lines!

File: spec/controllers/bathrooms_controller_spec.rb [hunk 4]
```diff
@@ -14,7 +14,69 @@
...
+#     let!(:params) { {bathroom: bath} }
+#     it "recieves bathroom data and creates a new bathroom" do
+#       post :create, params: params
+#
+#       bathroom = Bathroom.last
+#       expect(post :create, params: params).to change(Bathroom, :count).by(1)
+#       # expect(response.status).to eq 200
+#       # expect(response.content_type).to eq("application/json")
+#     end
+#   end
```
Comment (chrissyfancy) [inline spec/controllers/bathrooms_controller_spec.rb:81]: Is there a reason this is commented out?

File: app/controllers/api/v1/bathrooms_controller.rb (hunk 1 above)
Comment (nickbreid) [reply app/controllers/api/v1/bathrooms_controller.rb:30]: We haven't gotten to authorization stuff yet. Just wanted to get it working for anyone first

File: app/javascript/react/src/containers/BathroomForm.js (hunk 2 above)
Comment (nickbreid) [reply app/javascript/react/src/containers/BathroomForm.js:160]: Noted! We are planning to refactor this next week

File: spec/controllers/bathrooms_controller_spec.rb (hunk 4 above)
Comment (nickbreid) [reply spec/controllers/bathrooms_controller_spec.rb:81]: Not really. It's gone now!

**Pull request merged**